### PR TITLE
Implemented argparse; added some functions; changed tabs to spaces.

### PR DESCRIPTION
**Pull request by A-Vaillant**
The tabs to spaces thing is, arguably, an aesthetic difference but it _is_ PEP8. Adding in argparse lets us change the RC file loading from running a system command. I tried to keep it essentially identical in functionality to the  original version but it's fairly easy to, say, jettison the requirement for `none` when we want to do things without using a theme.

### CODE REVIEW

**Review by lasers — COMMENTED** [review]
Just passing through. Saw an issue in `--help`. Rofi is a nice touch. 
EDIT: You didn't add Rofi. Good thing we have `argparse` now to add missing args.

File: j4-make-config [hunk 1]
```diff
@@ -61,184 +57,177 @@ Append additional configuration and reload i3 after creating the config file:
...
-	print(info)
+    return info
+
+def printUsageInfo():
+    print(UsageInfo())
+
+def ThemeListStr():
+    return ("\nAvailable themes:\n"
+            "=================\n"
+            "\n".join(sorted(themes_list.keys())))
```
[the remark this answers is not in the document]
Comment (D-Vaillant) [reply j4-make-config]: Good catch!

File: j4-make-config [hunk 2]
```diff
@@ -61,184 +57,182 @@ Append additional configuration and reload i3 after creating the config file:
...
+            for theme in listdir(THEME_PATH):
+                themes_list[theme] = THEME_PATH + sep + theme
+        except (IOError, OSError):
+            pass
+
+        if themes_list == {}:
+            print("Error: No themes found.")
+            exit(1)
+
+    parser = argparse.ArgumentParser(description='i3 config generator/switcher',
```
Comment (lasers) [inline j4-make-config]: `description` might be better left unset because if you looked in `--help`, you'll see two of them. The other one is in `UsageInfo()`. With `argparse` in this PR, maybe the options should be printed correctly instead of faking it?

Comment (D-Vaillant) [reply j4-make-config]: Good suggestion. As it stands I can't think of a good way to show the usage information; it feels like it'd be better for a man page instead of the command line help. I added a "--themes" argument so we can preserve that but a "--usage" argument is weird to me.

A pet peeve: I'd like the arguments to be ordered so that '--themes' isn't put as the second argument but that's kind of finicky and requires knowledge of `argparse` internals that I do not have right now.

Comment (lasers) [reply j4-make-config]: I sent you a PR. You can continue working on top of that... enhancing `j4-make-config` further. :-)

This PR might alleviate some of your issues with `j4-make-config`. With too many themes, the list of themes could be hard to parse so I append them all on same line + 8 themes equals an additional line. Not sure if it's a good idea, but you can see the differences.

With `--list-themes` (and any other arguments, really), argparse permits least unambiguous prefix so we can just use `j4-make-config --list` too.

**Comment by lasers** [comment]
Bug.

Expected:
```shell
  ~/src/j4-make-config   master  ./j4-make-config
Executing: ./j4-make-config none
  ~/src/j4-make-config   master  ./j4-make-config
Executing: ./j4-make-config none
  ~/src/j4-make-config   master  ./j4-make-config
Executing: ./j4-make-config none
  ~/src/j4-make-config   master  ./j4-make-config
```
Got:
```shell
  ~/src/j4-make-config   D-Vaillant-argparse  ./j4-make-config none
  ~/src/j4-make-config   D-Vaillant-argparse  ./j4-make-config
Executing: ./j4-make-config none
  ~/src/j4-make-config   D-Vaillant-argparse  ./j4-make-config
Executing: ./j4-make-config
Error: theme argument is missing.
 ✘  ~/src/j4-make-config   D-Vaillant-argparse  ./j4-make-config
Executing: ./j4-make-config
Error: theme argument is missing.
 ✘  ~/src/j4-make-config   D-Vaillant-argparse  ./j4-make-config
```

**Comment by D-Vaillant** [comment]
Okay, I thought that one was gonna be harder to solve but it wasn't too bad. Eventually I need to make a testing suite for this.

**Comment by thiagowfx** [comment]
Hey Oliver, any chance you could add `D-Vaillant` as contributor of this project? (i.e. give them write access)
This seems like an useful pull request.

Another useful one is https://github.com/smheidrich/j4-make-config/commit/1247f7d81f97631fdfb7ff5c02441158e140f707.

Alternatively I'd be willing to merge/test those if you give me write access. That would probably better than me creating yet another fork of this repository :-)

All the best,

**Comment by D-Vaillant** [comment]
I forked this project into https://github.com/D-Vaillant/i3configs.
